### DISCUSSION TITLE
Why does it require us to scan the QR code?

### DISCUSSION THREAD

**Discussion by neoyk10** [Johnserf-Seed/TikTokDownload#494] [General]
Hello Sir. I updated to the latest version and it requires me to scan a QR code using Dou Yin app but I do not have a Dou Yin account. Could you fix this?

**Comment by JoeanAmier** [comment]
You must obtain a cookie, which you can obtain through the webpage

**Reply by neoyk10** [reply]
Could you elaborate a little further?

**Reply by JoeanAmier** [reply]
Please try filling this content in the `cookie` location of `conf. ini`

`passport_csrf_token=0a91351b367a22f52d4b91bb321f89c3; passport_csrf_token_default=0a91351b367a22f52d4b91bb321f89c3; csrf_session_id=a19636b03d7e61d7d03934adfa10c13f; odin_tt=9a196acafd67e5d0d865cd98d0e179154c2d9d99f1efe8b3b9296efc785123598aec079d672273a10a7170e8d8043cc415dd6fec5e425500792d154fc2adada69afaed1a15a64aaa5753fae398d0f697`

**Reply by JoeanAmier** [reply]
If it cannot be used, you can only register one Douyin account

**Reply by neoyk10** [reply]
I tried. It cannot be used still. 

I would love to but Dou Yin is no longer friendly to foreigners since it requires real name verification.

**Reply by menostech** [reply]
Traceback (most recent call last):
  File "F:\Douyin\TikTokDownload\tkd.py", line 12, in <module>
    profile = Util.Profile()

TypeError: Profile.__init__() missing 2 required positional arguments: 'config' and 'dyheaders'

I constructed the cookie in conf.ini, but new problem came out when I was running the code, how do I fix it?

**Comment by neoyk10** [comment]
How do I run this?

import Util

if __name__ == '__main__':
    # Get command line and configuration file
    cmd = Util.Command()
    config = cmd.config_dict
    dyheaders = cmd.dyheaders

    # Asynchronously download content
    Util.asyncio.run(Util.Profile(config, dyheaders).get_Profile())
    input("[  Prompt  ]: Download completed, press any key to exit.")

**Reply by sean0157** [reply]
You can create a Python file end up with  .py, and then copy all of these code into this  .py file, and then just run it
